From 32213f2f1d98bf851a570ecc35c018001e5d5ac4 Mon Sep 17 00:00:00 2001
From: Aaro Koskinen <aaro.koskinen@nokia.com>
Date: Mon, 23 Feb 2015 16:01:31 +0200
Subject: [PATCH] systemd: add getrandom syscall numbers for MIPS

Add getrandom syscall numbers for MIPS. Based on Linux 3.17 kernel
(commit 42944521af97a3b25516f15f3149aec3779656dc, "MIPS: Wire up new
syscalls getrandom and memfd_create").

(cherry picked from commit 3bec6d4690d2a7f08dc27b8221299c1db94978c4)
---
 src/shared/missing.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/shared/missing.h b/src/shared/missing.h
index 06a5576..8cb0b2c 100644
--- a/src/shared/missing.h
+++ b/src/shared/missing.h
@@ -243,6 +243,16 @@ static inline int memfd_create(const char *name, unsigned int flags) {
 #    define __NR_getrandom 349
 #  elif defined(__powerpc__)
 #    define __NR_getrandom 359
+#  elif defined _MIPS_SIM
+#    if _MIPS_SIM == _MIPS_SIM_ABI32
+#      define __NR_getrandom 4353
+#    endif
+#    if _MIPS_SIM == _MIPS_SIM_NABI32
+#      define __NR_getrandom 6317
+#    endif
+#    if _MIPS_SIM == _MIPS_SIM_ABI64
+#      define __NR_getrandom 5313
+#    endif
 #  else
 #    warning "__NR_getrandom unknown for your architecture"
 #    define __NR_getrandom 0xffffffff
